nexus/reconfigurator/planning/src/example.rs

```diff
@@ -23,12 +27,126 @@ use omicron_uuid_kinds::SledKind;
+        self.system_rng_gen += 1;
+        if self.system_rng_gen == 1 {
+            ExampleSystemRng::from_seed(self.seed.as_str())
+        } else {
+            ExampleSystemRng::from_seed((
+                self.seed.as_str(),
+                self.system_rng_gen,
+            ))
+        }
+    }
+
+    pub fn next_collection_rng(&mut self) -> CollectionBuilderRng {
+        self.collection_rng_gen += 1;
+        // We don't need to pass in extra bits unique to collections, because
+        // `CollectionBuilderRng` adds its own.
+        CollectionBuilderRng::from_seed((
+            self.seed.as_str(),
+            self.collection_rng_gen,
+        ))
+    }
+
+    pub fn next_planner_rng(&mut self) -> PlannerRng {
+        self.planner_rng_gen += 1;
+        // We don't need to pass in extra bits unique to the planner, because
+        // `PlannerRng` adds its own.
+        PlannerRng::from_seed((self.seed.as_str(), self.planner_rng_gen))
+    }
+
+    pub fn next_sled_id_rng(&mut self) -> TypedUuidRng<SledKind> {
+        self.sled_id_rng_gen += 1;
+        TypedUuidRng::from_seed(
+            self.seed.as_str(),
+            ("sled-id-rng", self.sled_id_rng_gen),
+        )
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ExampleSystemRng {
+    seed: String,
+    sled_rng: TypedUuidRng<SledKind>,
+    collection_rng: CollectionBuilderRng,
+    // ExampleSystem instances generate two blueprints: create RNGs for both.
+    blueprint1_rng: PlannerRng,
+    blueprint2_rng: PlannerRng,
+}
+
+impl ExampleSystemRng {
+    pub fn from_seed<H: Hash + fmt::Debug>(seed: H) -> Self {
+        // This is merely "ExampleSystem" for backwards compatibility with
+        // existing test fixtures.
+        let sled_rng = TypedUuidRng::from_seed(&seed, "ExampleSystem");
+        // We choose to make our own collection and blueprint RNGs rather than
+        // passing them in via `SimRngState`. This means that `SimRngState` is
+        // influenced as little as possible by the specifics of how
+        // `ExampleSystem` instances are generated, and RNG stability is
+        // maintained.
+        let collection_rng = CollectionBuilderRng::from_seed((
+            &seed,
+            "ExampleSystem collection",
+        ));
+        let blueprint1_rng =
+            PlannerRng::from_seed((&seed, "ExampleSystem initial"));
+        let blueprint2_rng =
+            PlannerRng::from_seed((&seed, "ExampleSystem make_zones"));
+        Self {
+            seed: format!("{:?}", seed),
+            sled_rng,
+            collection_rng,
+            blueprint1_rng,
+            blueprint2_rng,
+        }
+    }
+}
+
 /// An example generated system, along with a consistent planning input and
 /// collection.
 ///
 /// The components of this struct are generated together and match each other.
 /// The planning input and collection represent database input and inventory
-/// that would be collected from a system matching the system description.
```

Comment:
I think this line got dropped by accident?

Reply:
Thanks, fixed!